src/js/DropContainer.jsx

```diff
@@ -0,0 +1,281 @@
+/*eslint consistent-this: [2, "that"]*/
+
+import React from 'react';
+import classNames from 'classnames';
+import accepts from 'attr-accept';
+import $ from 'jquery';
+
+import css from '../stylesheets/button.css';
+import UploadButton from './UploadButton';
+import ImagePreview from './ImagePreview';
+
+export default class DropContainer extends React.Component {
+   /**
+      * Base Container
+      * @param  {Object} props
+   */
+   constructor(props) {
+      super(props);
+      this.state = {
+         imageFile: '',
+         isDragActive: false,
+         borderStyle: '',
+         visibility: 'show'
+      };
+      this.handleChange = this.handleChange.bind(this);
+      this.onDragOver = this.onDragOver.bind(this);
+      this.onDrop = this.onDrop.bind(this);
+      this.onDragLeave = this.onDragLeave.bind(this);
+      this.onDragEnter = this.onDragEnter.bind(this);
+      this.onDragStart = this.onDragStart.bind(this);
+      // this.onClick = this.onClick.bind(this);
+   }
+
+   componentDidMount() {
+      this.enterCounter = 0;
+      // Change the background color of
+      // outer section with the prop color
+      // passing in
+      let $this = $(this.myTextInput)[0],
+         color = this.props.backgroundColor;
+      $($this).css('background-color', color);
+   }
+
+   onDragStart(e) {
+      if (this.props.onDragStart) {
+         this.props.onDragStart.call(this, e);
+      }
+   }
+
+   onDragEnter(e) {
+      e.preventDefault();
+
+      // Count the dropzone and any children that are entered.
+      ++this.enterCounter;
+
+      // This is tricky. During the drag even the dataTransfer.files is null
+      // But Chrome implements some drag store, which is accesible via dataTransfer.items
+      const dataTransferItems = this.extractData(e);
+
+      // Now we need to convert the DataTransferList to Array
+      const allFilesAccepted = this.fileAccepted(Array.prototype.slice.call(dataTransferItems));
+
+      this.setState({
+         isDragActive: allFilesAccepted,
+         borderStyle: 'dragBorder',
+         visibility: 'hide'
+         // isDragReject: !allFilesAccepted
+      });
+
+      if (allFilesAccepted) {
+         this.successBorder()
+      } else {
+         this.warningBorder();
+      }
+  }
+
+   /**
+      * Handle the drag over event
+      * @param  {Event]} event
+   */
+   onDragOver (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      e.dataTransfer.dropEffect = 'copy';
+
+      return false;
+   }
+
+   /**
+      * Handle the drag leave event
+   */
+   onDragLeave (e) {
+      e.preventDefault();
+
+      // Only deactivate once the dropzone and all children was left.
+      if (--this.enterCounter > 0) {
+         return;
+      }
+
+      this.setState({
+         isDragActive: false,
+         isDragReject: false,
+         borderStyle: '',
+         visibility: 'show'
+      });
+
+      if (this.props.onDragLeave) {
+         this.props.onDragLeave.call(this, e);
```

Comment:
Same issue with call as before. Best to not use call/apply on them if you can help it